app/Config/Database.php

```diff
@@ -25,23 +25,24 @@ class Database extends Config
      * The default database connection.
      */
     public array $default = [
-        'DSN'      => '',
-        'hostname' => 'localhost',
-        'username' => '',
-        'password' => '',
-        'database' => '',
-        'DBDriver' => 'MySQLi',
-        'DBPrefix' => '',
-        'pConnect' => false,
-        'DBDebug'  => true,
-        'charset'  => 'utf8',
-        'DBCollat' => 'utf8_general_ci',
-        'swapPre'  => '',
-        'encrypt'  => false,
-        'compress' => false,
-        'strictOn' => false,
-        'failover' => [],
-        'port'     => 3306,
+        'DSN'          => '',
+        'hostname'     => 'localhost',
+        'username'     => '',
+        'password'     => '',
+        'database'     => '',
+        'DBDriver'     => 'MySQLi',
+        'DBPrefix'     => '',
+        'pConnect'     => false,
+        'DBDebug'      => true,
+        'charset'      => 'utf8',
+        'DBCollat'     => 'utf8_general_ci',
+        'swapPre'      => '',
+        'encrypt'      => false,
+        'compress'     => false,
+        'strictOn'     => false,
+        'failover'     => [],
+        'port'         => 3306,
+        'DBTypeNative' => false,
```

Comment:
I don't like the name you have chosen. This would be my suggestion, but I would like to hear from others about it... maybe `nativeType`... @kenjis?

```suggestion
        'strictType' => false,
```

Reply:
I don't like `DBTypeNative`. `strictType` seems a bit different.

`intAndFloatNative`. `numberNative`. Too specific?
`nativeType` may be good if we prefer generic name.

Reply:
+1 for `numberNative` - this is only for MySQL so we should be good.

Reply:
Thank you both for your replies. 
I'll change the variable name to numberNative.

Reply:
Sorry to interrupt your time, @michalsn .
I would like to ask about the test, if I move `numberNative `to Connect.php in MySQLi, can I do the data type test directly in RawSqlTest.php? Or do I need to do the test for this property in ConnectTest.php separately?

Reply:
Personally, I think the best place for these tests will be in `Database/Live/MySQLi/NumberNativeTest.php` (new file).

You can look at https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/system/Database/Live/TransactionDBDebugTrueTest.php - as a reference of how you can build these tests - it might be a lot better example than I suggested before.

Make just one test for selecting number values when `numberNative` is enabled and one more when this option is disabled. That's all we need.

Reply:
Crap... sorry, that won't be a better example. We need to set the correct config before we connect to the database.

Reply:
I have added a test file, the content of some simple tests, if there is a need to add I will fill in.